llvm/test/tools/sycl-post-link/spec-constants/spec_const_and_split.ll

```diff
@@ -2,27 +2,60 @@
 ; requested.
 ;
+; Additionally this test checks that sycl-post-link creates specialization
+; constant property correctly: i.e. it doesn't put all specialization constants
+; into properties of each device image
+
 ; RUN: sycl-post-link -split=kernel -spec-const=rt -S %s -o %t.files.table
-; RUN: FileCheck %s -input-file=%t.files_0.ll --check-prefixes CHECK0,CHECK
-; RUN: FileCheck %s -input-file=%t.files_1.ll --check-prefixes CHECK1,CHECK
+; RUN: FileCheck %s -input-file=%t.files_0.ll --check-prefixes CHECK-IR0,CHECK-IR
+; RUN: FileCheck %s -input-file=%t.files_1.ll --check-prefixes CHECK-IR1,CHECK-IR
+; RUN: FileCheck %s -input-file=%t.files_2.ll --check-prefixes CHECK-IR2,CHECK-IR
+; RUN: FileCheck %s -input-file=%t.files_0.prop --check-prefixes CHECK-PROP0
+; RUN: FileCheck %s -input-file=%t.files_1.prop --check-prefixes CHECK-PROP1
+; RUN: FileCheck %s -input-file=%t.files_2.prop --check-prefixes CHECK-PROP2
 
 @SCSymID = private unnamed_addr constant [10 x i8] c"SpecConst\00", align 1
-; CHECK-NOT: @SCSymID
+@SCSymID2 = private unnamed_addr constant [11 x i8] c"SpecConst2\00", align 1
+; CHECK-IR-NOT: @SCSymID
+; CHECK-IR-NOT: @SCSymID2
 
 declare dso_local spir_func zeroext i1 @_Z33__sycl_getScalarSpecConstantValueIbET_PKc(i8 addrspace(4)*)
 
 define dso_local spir_kernel void @KERNEL_AAA() {
   %1 = call spir_func zeroext i1 @_Z33__sycl_getScalarSpecConstantValueIbET_PKc(i8 addrspace(4)* addrspacecast (i8* getelementptr inbounds ([10 x i8], [10 x i8]* @SCSymID, i64 0, i64 0) to i8 addrspace(4)*))
-; CHECK0: %{{[0-9]+}} = call i1 @_Z20__spirv_SpecConstantib(i32 0, i1 false)
+; CHECK-IR0: %{{[0-9]+}} = call i1 @_Z20__spirv_SpecConstantib(i32 1, i1 false)
+  %2 = call spir_func zeroext i1 @_Z33__sycl_getScalarSpecConstantValueIbET_PKc(i8 addrspace(4)* addrspacecast (i8* getelementptr inbounds ([11 x i8], [11 x i8]* @SCSymID2, i64 0, i64 0) to i8 addrspace(4)*))
+; CHECK-IR0: %{{[0-9]+}} = call i1 @_Z20__spirv_SpecConstantib(i32 0, i1 false)
   ret void
 }
 
 define dso_local spir_kernel void @KERNEL_BBB() {
   %1 = call spir_func zeroext i1 @_Z33__sycl_getScalarSpecConstantValueIbET_PKc(i8 addrspace(4)* addrspacecast (i8* getelementptr inbounds ([10 x i8], [10 x i8]* @SCSymID, i64 0, i64 0) to i8 addrspace(4)*))
-; CHECK1: %{{[0-9]+}} = call i1 @_Z20__spirv_SpecConstantib(i32 0, i1 false)
+; CHECK-IR1: %{{[0-9]+}} = call i1 @_Z20__spirv_SpecConstantib(i32 0, i1 false)
+  ret void
+}
+
+define dso_local spir_kernel void @KERNEL_CCC() {
+; CHECK-IR2: define{{.*}}spir_kernel void @KERNEL_CCC
   ret void
 }
 
-; CHECK: !sycl.specialization-constants = !{![[#MD:]]}
+; CHECK-IR0: !sycl.specialization-constants = !{![[#MD0:]], ![[#MD1:]]}
+; CHECK-IR0: ![[#MD0:]] = !{!"SpecConst2", i32 0, i32 0, i32 1}
+; CHECK-IR0: ![[#MD1:]] = !{!"SpecConst", i32 1, i32 0, i32 1}
+;
+; CHECK-IR1: !sycl.specialization-constants = !{![[#MD0:]]}
+; CHECK-IR1: ![[#MD0:]] = !{!"SpecConst", i32 0, i32 0, i32 1}
+;
+; CHECK-IR2: !sycl.specialization-constants = !{}
```

Comment:
I think that ideally, we shouldn't generate this metadata at all and therefore we shouldn't generate corresponding property set (even if its empty), but I would like to address this in a separate PR

Reply:
Personally, I don't mind.